src/s11/handlers/create_session_handler.c

```diff
@@ -251,10 +251,15 @@ create_session_processing()
 
 
 	ies[ieCount].type = IE_APN;
-	ies[ieCount].length = htons(g_csReqInfo->apn.len);
+	ies[ieCount].length = htons((int)(g_csReqInfo->selected_apn.len) + 1);
+	log_msg(LOG_INFO, "g_csReqInfo->selected_apn.len = %d\n",
+			(int)(g_csReqInfo->selected_apn.len));
 	ies[ieCount].instance = INSTANCE_ZERO;
-	memcpy(ies[ieCount].value, g_csReqInfo->apn.val,
-			g_csReqInfo->apn.len);
+	unsigned char apn_len[25] = {};
```

Comment:
Using unsigned char is good enough. No need to use array.
e.g. unsigned char app_len;
app_len = g_csReqInfo->selected_apn.len;

Reply:
the apn_len it self is no need an array.
later on it is used in strcat(apn_len, g_csReqInfo->selected_apn.val), this is why make it as array, the name is a confusing. I will change it.

Reply:
fixed in paging branch.